app/frontend/src/cljs/teet/asset/materials_and_products_view.cljs

```diff
@@ -77,13 +102,39 @@
                     :fgroup-link-fn filter-link-fn
                     :list-features? false}}
        [:div.cost-items-totals
+        [asset-ui/filter-breadcrumbs {:atl atl
+                                      :root-label (tr [:asset :totals-table :all-materials])
+                                      :query query
+                                      :filter-kw filter-fg-or-fc
+                                      :page :materials-and-products}]
         [:div {:style {:max-width "25vw"}}
          [typography/Heading1 "Materials"]]
         [table/listing-table-container
          [table/listing-header (assoc listing-opts :state listing-state)]
-         [table/listing-body (assoc listing-opts
-                                    :rows materials-and-products
-                                    )]]]])))
+         (doall
+          (for [[fg fgroup-rows] (->> filtered-materials-and-products
+                                      (group-by (comp first :ui/group))
+                                      ;; sort by translated fgroup label
+                                      (sort-by (comp asset-ui/label first)))
+                :let [ident (:db/ident fg)
+                      open? (not (closed-sections ident))]]
+            ^{:key (str ident)}
+            [:<>
+             [table-section-header e! query listing-opts closed-sections fg nil]
+             (when open?
+               [:<>
+                (doall
+                 (for [[fc fclass-rows] (group-by (comp second :ui/group)
+                                                  fgroup-rows)
+                       :let [ident (:db/ident fc)
+                             open? (not (closed-sections ident))]]
+                   ^{:key (str ident)}
+                   [:<>
+                    [table-section-header e! query listing-opts closed-sections fc nil]
+                    (when open?
+                      [table/listing-body (assoc listing-opts
+                                                 :key (comp str #(get-column atl % :parameter))
+                                                 :rows fclass-rows)])]))])]))]]])))
```

Comment:
Now that I have a function that produces `:ui/group` for materials as well, this is also mostly repetition from `cost-groups-view`.